### PR TITLE
[Bugfix] Preserve non-logitproc entry points in tests

### PR DESCRIPTION
**Pull request by d4l3k**
## Purpose

The fork-path logits processor test helper replaces `importlib.metadata.entry_points` process-wide and previously returned an empty result for every group other than `vllm.logits_processors`. PyTorch nightly now registers built-in distributed backends through `torch.distributed.backends`, so forked engine workers could not discover NCCL and failed with `Unknown c10d backend type NCCL`.

Delegate non-logitproc queries, including the no-argument metadata API, to the original implementation while continuing to inject the fake logits processor group. Add a focused regression test that covers both behaviors.

This addresses the vLLM failure reported in https://github.com/pytorch/pytorch/issues/192062.

Duplicate check: searched open vLLM issues and PRs for the PyTorch issue number, the NCCL error, `torch.distributed.backends`, and logits processor entry-point fixes; no matching work was open.

AI assistance: OpenAI Codex assisted with diagnosis, implementation, and testing. The human submitter reviewed the changed lines and test results.

Model evaluation: Not applicable because this only changes test entry-point mocking and does not affect serving or model output.

## Test Plan

- `uv run --no-sync --python .venv/bin/python python -m pytest tests/v1/logits_processors/test_custom_offline.py -k fake_entrypoint_preserves_other_groups -v`
- `CUDA_VISIBLE_DEVICES=2 uv run --no-sync --python .venv/bin/python python -m pytest tests/v1/logits_processors/test_custom_offline.py::test_custom_logitsprocs[CustomLogitprocSource.LOGITPROC_SOURCE_ENTRYPOINT] -v -s`
- `CUDA_VISIBLE_DEVICES=2 uv run --no-sync --python .venv/bin/python python -m pytest tests/v1/logits_processors/test_custom_offline.py -k "test_rejects_custom_logitsprocs and LOGITPROC_SOURCE_ENTRYPOINT" -v`
- `uv run --no-sync --python .venv/bin/python pre-commit run --files tests/v1/logits_processors/utils.py tests/v1/logits_processors/test_custom_offline.py`

## Test Result

All commands passed with `torch 2.14.0.dev20260804+cu130`. The nightly end-to-end entry-point test passed both fresh NCCL engine initializations and generation; the two rejection-path variants also passed.

### CODE REVIEW

**Review by atalman — APPROVED** [review]
Thank you!

**Review by zou3519 — APPROVED** [review]
